base_agent/memory_nodes.py

```diff
@@ -789,17 +807,92 @@ def create(cls, memory, task: Task) -> str:
             >>> task = Task()
             >>> create(memory, task)
         """
+        old_memid = getattr(task, "memid", None)
+        if old_memid:
+            return old_memid
         memid = cls.new(memory)
         task.memid = memid  # FIXME: this shouldn't be necessary, merge Task and TaskNode?
-        memory.db_write(
-            "INSERT INTO Tasks (uuid, action_name, pickled, created_at) VALUES (?,?,?,?)",
+        memory._db_write(
+            "INSERT INTO Tasks (uuid, action_name, pickled, prio, running, created_at) VALUES (?,?,?,?,?,?)",
             memid,
             task.__class__.__name__,
             memory.safe_pickle(task),
+            task.prio,
+            task.running,
             memory.get_time(),
         )
         return memid
 
+    def step(self, agent):
+        self.task.step(agent)
+        self.update_task()
+
+    def update_task(self, task=None):
+        task = task or self.task
+        self.memory.db_write(
+            "UPDATE Tasks SET pickled=? WHERE uuid=?", self.memory.safe_pickle(task), self.memid
+        )
+
+    # FIXME TODO don't need paused, set prio to 0 and have a condtion for unpausing
+    # use this to update prio or running, don't do it directly on task or in db!!
+    def get_update_status(self, status, force_db_update=True, force_task_update=True):
```

Comment:
Is this name reflective of what the function does? From the name it seems like the function just does read operations (gets statues) but from the docstring it looks like the function does a lot of updates as well.

Reply:
I can change the name; it _gets_ and _updates_ (so get_update)